.github/workflows/cpp_format_tools.yml

```diff
@@ -0,0 +1,34 @@
+name: C++ Format Tools Docker Image
+on:
+  push:
+    tags: [ '**' ]
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+    paths:
+      - .github/workflows/cpp_format_tools.yml
+      - cpp_format_tools/Dockerfile
+      - cpp_format_tools/format.sh
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build the Docker image
+      run: docker build cpp_format_tools/. -t cpp_format_tools
+    - name: Push the Docker image
+      if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
+      run: |
+        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
```

Comment:
may want to use `docker/login-action@v1`

Reply:
Done.